src/QirRuntime/lib/QIR/intrinsics.cpp

```diff
@@ -56,55 +97,102 @@ extern "C"
         return Microsoft::Quantum::g_context->simulator->M(qubit);
     }
 
-    void quantum__qis__s(Qubit qubit)
+    void quantum__qis__r__body(PauliId axis, double angle, QUBIT* qubit)
+    {
+        return Microsoft::Quantum::g_context->simulator->AsQuantumGateSet()->R(axis, qubit, angle);
+    }
+
+    void quantum__qis__r__adj(PauliId axis, double angle, QUBIT* qubit)
+    {
+        return Microsoft::Quantum::g_context->simulator->AsQuantumGateSet()->R(axis, qubit, -angle);
+    }
+
+    void quantum__qis__r__ctl(QirArray* ctls, PauliId axis, double angle, QUBIT* qubit)
+    {
+        return Microsoft::Quantum::g_context->simulator->AsQuantumGateSet()->ControlledR(
+            ctls->count, reinterpret_cast<Qubit*>(ctls->buffer), axis, qubit, angle);
+    }
+
+    void quantum__qis__r__ctladj(QirArray* ctls, PauliId axis, double angle, QUBIT* qubit)
+    {
+        return Microsoft::Quantum::g_context->simulator->AsQuantumGateSet()->ControlledR(
+            ctls->count, reinterpret_cast<Qubit*>(ctls->buffer), axis, qubit, angle);
+    }

```

Comment:
Minor:
I see that 
`quantum__qis__r__ctl()` and 
`quantum__qis__r__ctladj()` are identical.
May be it is possible to use a single function instead of two? Or can one of them just call the other of them, in such a way showing explicitly that they have the same implementation?

Reply:
Yeap, another bug :(